(16-qpr2) - Gradle, AndroidX, and a number of other fixes.

oscar-labit opened the pull request:
Sorry for the duplicate pull requests. I thought merging the 16-qpr2 branch with my own would take longer than it did.

Changes since PR #52 :
* Merged 16-qpr2 commits.

I'll continue testing to ensure there are no bugs over the next few weeks.

AdrianDC committed:
Dialer: disable anti-falsing for call answer screen

 * The anti-falsing implementation from HumanInteractionClassifier
    regularly prevents easy swipe to answer, requiring multiple
    attempts until accepted.

Change-Id: Iebad27f9da7bb8fea6fc663dc99ac6c17d94ed5f
(cherry picked from commit 583e42df0f6f02f1a6c05dbdc5559f42e99c8727)

bvlj committed:
Dialer: define app category

Change-Id: I53b6ee1c70925c50e91270e0b084d777dd8c06f6
Signed-off-by: Joey <joey@lineageos.org>

Commit:
Fix crash issue when user taps audio button while BT is connected

Since incallui doesn't have BLUETOOTH permission, SecurityException
occurs when using BluetoothDevice#getAliasName() API.
To resolve this issue, add the BLUETOOTH permission.

Test: manual - Checked that no crash occurred when switching to
audio route with BT connected.
Bug: 127236848

Change-Id: I65fa3d6275569e967d9608ac4b1c8ee175b2664d

codeworkx committed:
Dialer: AudioModeProvider: use wired route for usb headsets

Change-Id: I920a9c259af5597dfd9a88a72e4ac54cb6e73f7a

jesec committed:
Dialer: allow framework to do dark theming automatically

Change-Id: I5b840ba257adda69882e1fe19a55510802c45106
Signed-off-by: Jesse Chan <jc@lineageos.org>

Tortel committed:
Dialer: Use a DayNight launch theme

This prevents the white launch screen from appearing when the system theme
is dark

Change-Id: I1ee8ff493910578b5e5c9de67ea1f7390178051a

trautamaki committed:
Dialer: use black/white navigation colors

Change-Id: I9d2ee0bbc7d765e605d6f60c7866565c83dff295

trautamaki committed:
Dialer: update dark theme colors

The dark theme colors exist in
java/com/android/dialer/theme/hidden/res/values/colors_dialer_light.xml
that would be used with AOSPThemeModule. As the app uses framework
dark theme instead of the module, copy the dark theme colors to
values-night.

Change-Id: I02a36b01ba581e62556ce6e0151ff0a35209467a

trautamaki committed:
Dialer: update ripple color

Ripple on dark should be lighter than background.

Change-Id: I01993eac93f29609353ef1e1e9b5dc9a211a640c

trautamaki committed:
Dialer: set statusbar color same as activity background

 * Keep styles that use actionBar as is because it uses
   normal actionbar
 * Set light statusbar attribute for some styles

Change-Id: I8748aa4d12f8e0c68d14d514ea5a2bd4100ee3ca

trautamaki committed:
Dialer: improve search bar

 * Make background color lighter
 * Update border radius
 * Update ripple to follow border radius
 * Remove the background color, use same as activity
 * Update the back-icon color to white/black for consistency

Change-Id: I648a2bd0da47fec99d8a8457c70451e77ed9f8be

maniac103 committed:
Fix status bar colors in detail activities.

Change-Id: I0994bd1fd9922f18a08364a0d88426d88cb04b6a

flawedworld committed:
Add GrapheneOS icon

Commit:
Replace launcher icon

razorloves committed:
fixup! Add a permission to guard receiving intents in StatusSmsFetcher.

Add missing permission declaration.

Change-Id: Icc71b79831c3045b36ff43cc0884275c1c7cbbc8

theimpulson committed:
proguard.flags: Don't optimize fragment constructors

09-03 13:08:51.712  4496  4496 E MessageQueue-JNI: Exception in MessageQueue callback: handleReceiveCallback
09-03 13:08:51.713  4496  4496 E MessageQueue-JNI: android.app.Fragment$InstantiationException: Unable to instantiate fragment com.android.dialer.app.settings.DisplayOptionsSettingsFragment: calling Fragment constructor caused an exception
09-03 13:08:51.713  4496  4496 E MessageQueue-JNI:      at android.app.Fragment.instantiate(Fragment.java:565)
09-03 13:08:51.713  4496  4496 E MessageQueue-JNI:      at android.preference.PreferenceActivity.switchToHeaderInner(PreferenceActivity.java:1245)
09-03 13:08:51.713  4496  4496 E MessageQueue-JNI:      at android.preference.PreferenceActivity.switchToHeader(PreferenceActivity.java:1296)
09-03 13:08:51.713  4496  4496 E MessageQueue-JNI:      at android.preference.PreferenceActivity.onHeaderClick(PreferenceActivity.java:1093)
09-03 13:08:51.713  4496  4496 E MessageQueue-JNI:      at com.android.dialer.app.settings.DialerSettingsActivity.onHeaderClick(DialerSettingsActivity.java:45)

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>
Change-Id: Idef793065ea422e7506f943bc76cd20e60e0a271

empratyush committed:
add monochrome icon

x86pup committed:
Don't show homescreen wallpaper for incallui

Signed-off-by: r3g_5z <june@girlboss.ceo>

empratyush committed:
add dark mode to in call ui dialer

x86pup committed:
Implement call recording

Co-authored-by: Danny Baumann <dannybaumann@web.de>
Signed-off-by: r3g_5z <june@girlboss.ceo>

muhomorr committed:
callrec: remove unused call recording migration code

muhomorr committed:
callrec: never allow access to call recordings through the FileProvider

muhomorr committed:
callrec: remove no-op implementations of interfaces of call recording UI

muhomorr committed:
callrec: bugfix: default values of call recording preferences were ignored

muhomorr committed:
callrec: bugfix: file descriptor leak in CallRecorderService

MediaRecorder dups the file descriptor that is passed to it.
Leaked file descriptors were cleared out eventually by ParcelFileDescriptor finalizer.

muhomorr committed:
callrec: catch exceptions from reset() when releasing MediaRecorder

reset() can throw an exception, despite docs not mentioning it.

muhomorr committed:
callrec: don't ask for legacy WRITE_EXTERNAL_STORAGE permission

octocorvus committed:
[temporary] incallui: increase number of buttons in button grid to 12

This is achieved by making the whole button grid scrollable.

maade93791 committed:
CI: update VVM config from 121.0.603393336

muhomorr committed:
15: fix crashes caused by removal of Fragment constructors by r8

empratyush committed:
use app compat theme

neelchauhan committed:
Enable darkmode on dialogs

Closes https://github.com/GrapheneOS/os-issue-tracker/issues/4470

Commit:
Snap for 13147095 from 17323f4d36f7193f8f23ed553cac8184cec5236f to 25Q3-release

Change-Id: I971d02ac5bf57f8d14f8ca229334c8739eb2f674

Commit:
Snap for 13436862 from 768ed834ced557e2d42476757776a26c68fed656 to 25Q3-release

Change-Id: Iacb6da339bd144258e43d0d4d971d584bc1bddd9

inthewaves committed:
incallui: always show scrollbar for button grid

Scrollbar isn't shown if there aren't more than 6 buttons

inthewaves committed:
incallui: move RTT to the end and prioritize call recording button

Commit:
[vvm] fix for verizon MVNOs

- moves 3-4 verizon based MVNOs back to vvm_type_vvm3
- allows voicemail notifications & VVM to work for those carriers (like
  straighttalk/tracfone)

oscar-labit committed:
Initial working Gradle commit. Transitioned to AndroidX, merged duplicate resources, configured Protobuf, and fixed R classes.

oscar-labit committed:
AndroidX import and dependency corrections, updated libraries, osmand integration, and other fixes.

oscar-labit committed:
Fixed build errors. Added dev environment to test on physical GOS phones.

oscar-labit committed:
Fixed PendingIntent errors.

oscar-labit committed:
Replaced java imports with just java.

oscar-labit committed:
Javadoc reference errors, NotificationCompat errors, API 33 errors, API 34 errors, updated dynamic names for dev build.

oscar-labit committed:
Removed unused imports, fixed some package declaration, and replaced @TargetApi

oscar-labit committed:
Fixed formatting and API error

oscar-labit committed:
Fixed javadoc errors

oscar-labit committed:
Fixed app not showing up in default phone app, updated dev build name, and other misc fixes

oscar-labit committed:
Fixed call foreground services

oscar-labit committed:
Made voicemail always viewable, added voicemail permission request.

oscar-labit committed:
Fixed misc errors

oscar-labit committed:
Fixed image cache crash and made voicemail tab permanent

oscar-labit committed:
(BROKE) Fixed VVM crash. Need to finish fixing the notification decline/answer buttons.

oscar-labit committed:
Fixed misc IDE code warnings and errors: API versioning, redundant checks, null checks, deprecated checks, etc.

oscar-labit committed:
(FIX) merge java directories, fixed answer/decline dropdown notification

oscar-labit committed:
Fixed navigation bar appearing black on dark mode.

oscar-labit committed:
More comprehensive .gitignore

oscar-labit committed:
Merge remote-tracking branch 'GOS/16-qpr2' into merge-16qpr2

# Conflicts:
#	AndroidManifest.xml
#	java/com/android/dialer/app/res/values/colors.xml
#	java/com/android/dialer/app/settings/CallRecordingSettingsFragment.java
#	java/com/android/dialer/callrecord/impl/CallRecorderService.java
#	java/com/android/dialer/main/impl/res/values/styles.xml
#	java/com/android/incallui/CallButtonPresenter.java
#	java/com/android/incallui/call/CallRecorder.java
#	java/com/android/incallui/incall/impl/InCallButtonGridFragment.java

oscar-labit committed:
Fixed duplicate resource after merge

oscar-labit commented:
Converted to draft while I test for bugs.

oscar-labit committed:
Removed gradle-wrapper.jar

oscar-labit committed:
Added distributionSha256Sum

oscar-labit commented:
Known (I'll fix these after New Years): 
* Call history and settings buttons are broken.
* The other active build variants aren't working (release file).
* Repeated and redundant code bloating the size of the app .
* Voice search is unavailable (though it looks like this is an issue with 16-qpr2 anyway).

I'm also working on a Material Design 3 implementation while I test for bugs under [this forum post's](https://discuss.grapheneos.org/d/14323-plans-to-replace-or-overhaul-the-aosp-apps) approach number 2. I have the main app working fine now. I'm working up updating the in-call UI and the contacts view next.

oscar-labit committed:
Fixed settings and call history, replaced gradlew with Groovy template

oscar-labit committed:
Fixed build variants

oscar-labit committed:
Fixed the call history back button

oscar-labit commented:
I've tested just about every functionality. It all seems to be working fine.

oscar-labit commented:
Known issues:
* The missed call notification buttons (call back; message) are broken.
* Contacts load oddly after launching the app post-restart. I'm unsure if this is an issue with this PR.

I don't have time to work on this for a while. I'll update it when I can.

bentemple commented:
Working great for me, thank you for doing this. Release build crashes for voicemail and recent calls tab because of proguard messing up the protobuf generated files. I added the following and that resolved the issues:

```
diff --git a/proguard.flags b/proguard.flags
index 13607913a..6266e788a 100644
--- a/proguard.flags
+++ b/proguard.flags
@@ -35,3 +35,28 @@
 # Keep Dagger component implementations
 -keep class com.android.dialer.binary.aosp.DaggerAospDialerRootComponent { *; }
 -keep class com.android.dialer.binary.aosp.DaggerAospDialerRootComponent$* { *; }
+
+# Protobuf-lite rules - required for reflection-based field access
+# All protobuf-generated message classes extend GeneratedMessageLite
+-keep class * extends com.google.protobuf.GeneratedMessageLite { *; }
+-keep class * extends com.google.protobuf.GeneratedMessageLite$Builder { *; }
+
+# Keep inner classes (nested messages like CallDetailsEntries$CallDetailsEntry)
+-keep class * extends com.google.protobuf.GeneratedMessageLite$* { *; }
+
+# Keep OrBuilder interfaces used by protobuf
+-keep interface * extends com.google.protobuf.MessageLiteOrBuilder { *; }
+
+# Keep protobuf enums
+-keepclassmembers class * extends com.google.protobuf.Internal$EnumLite {
+    public static **[] values();
+    public static ** valueOf(int);
+}
+
+# Keep OuterClass containers generated by protobuf
+-keep class **OuterClass { *; }
+-keep class **OuterClass$* { *; }
+
+# gRPC rules (used with protobuf)
+-keep class io.grpc.** { *; }
+-dontwarn io.grpc.**
```

oscar-labit committed:
Protobuf updates to avoid build crashes

oscar-labit commented:
@bentemple I added the changes. Please let me know if you come across anything else.

I removed the MD3 commits because those weren't meant to be added to this branch. See the draft PR #54 for those changes.
